script.js

```diff
@@ -1,4 +1,208 @@
+
+//global variables
+var DICEROLL = `DICEROLL`;
+var DICEORDER = `DICEORDER`;
+var gameMode = DICEROLL;
+var player1Dice = [];
+var player2Dice = [];
+var currPlayer = 1;
+var player1Num = [];
+var player2Num = [];
+var player1SumNum = [];
+var player2SumNum = [];
+var player1TotalScore;
+var player2TotalScore;
+
+// Possible Modes
+var NORMALMODE = `Normal Mode`;
+var LOWESTCOMBINEDNUMER = `Lowest Combined Number Mode`;
+
+// Mode Selection 
+var MODE_SELECTION_INSTRUCTIONS = 
+'Please enter one of the following to choose a game mode:<br>1. ' +
+NORMALMODE +
+'<br>2. ' +
+LOWESTCOMBINEDNUMER +`.`
+
+// keep track of game mode
+var primaryGameMode = ``
+
+var rollDice = function () {
+  return Math.ceil(Math.random() * 6)
+}
+
+var getDiceRolls = function () {
+  var diceRolls = [rollDice(), rollDice()]
+  
+  if (currPlayer == 1) {
+    player1Dice = diceRolls;
+  } else 
+    {
+    player2Dice = diceRolls;
+    }
+    return diceRolls;
+}
+
+var concatenate2Numbers = function (num1, num2) {
+  return Number(String(num1) + String(num2));
 };
+
+
+  var getPlayerNumber = function (diceOrder) {
+  
+    var diceArray;
+    if (currPlayer == 1) {
+        diceArray = player1Dice
+        } else
+        {diceArray = player2Dice};
+
+    var playerNum;
+
+    if (diceOrder === 1) {
+      playerNum = concatenate2Numbers(diceArray[0], diceArray[1]);
+    }
+    else {
+      playerNum = concatenate2Numbers(diceArray[1], diceArray[0]);
+    }
+  
+    if (currPlayer === 1) {
+      player1Num = playerNum;
+    } else {
+      player2Num = playerNum;
+    }
+  
+    return playerNum;
+  };
+
+
+var calcTotalNumber = function(array) {
+  
+  let sum = 0;
+  for (let i =0; i <array.length;i++) {
+    console.log(`array[i]`)
+    console.log(array[i])
+    sum += array[i];
+
+  }
+  console.log(sum);
+  return sum;
+}
+
+var determineWinner = function() {
+
+  if (primaryGameMode == NORMALMODE && player1Num > player2Num ||
+    primaryGameMode == LOWESTCOMBINEDNUMER && player1Num < player2Num) {
+    return 1;}
+    
+    return 2;
+  
+  };
+
+
+var leaderBoard =  function () {
+  if (player1TotalScore > player2TotalScore) {
+    return `Player 1 is in the Lead!<br><br>
+    Player 1 Total Score ${player1TotalScore}<br><br>
+    Player 2 Total Score ${player2TotalScore}`
+  } else
+   return `Player 2 is in the Lead!<br><br>
+   Player 2 Total Score ${player2TotalScore}<br><br>
+   Player 1 Total Score ${player1TotalScore}`
+}
+
+var lowestCombinedNumberBoard = function () {
+  if (player1TotalScore < player2TotalScore) {
+    return `Player 1 is in the Lead!<br><br>
+    Player 1 Total Score ${player1TotalScore}<br><br>
+    Player 2 Total Score ${player2TotalScore}`
+  } else
+   return `Player 2 is in the Lead!<br><br>
+   Player 2 Total Score ${player2TotalScore}<br><br>
+   Player 1 Total Score ${player1TotalScore}`
+}
+
+var main = function(input) {
```

Comment:
Do consider adding more comments to explain what you're trying to achieve in each line of code, which may be helpful when you try to reread your code in future.